lib/ide/src/view.rs

```diff
@@ -13,3 +13,45 @@ pub mod temporary_panel;
 pub mod project;
 pub mod layout;
 pub mod text_editor;
+
+
+
+// ========================
+// === KeyboardListener ===
+// ========================
+// This code is temporary and will be replace with FRP keyboard events once it's fully functional.
+
+use wasm_bindgen::prelude::*;
+use basegl::prelude::*;
+
+use basegl::system::web::document;
+use js_sys::Function;
+use web_sys::HtmlElement;
+use web_sys::KeyboardEvent;
+use wasm_bindgen::JsCast;
+
+
+type KeyboardClosure = Closure<dyn FnMut(KeyboardEvent)>;
+
+#[derive(Debug)]
+struct KeyboardListener {
+    callback   : KeyboardClosure,
+    element    : HtmlElement,
+    event_type : String
+}
+
+impl KeyboardListener {
```

Comment:
2 questions:

1. Why KeyboardListener is defined in `view`?
2. What is the relation between this keyboard listener and the one created by Adam which is bound to FRP?